lib/publiccloud/instance.pm

```diff
@@ -91,4 +95,96 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+sub wait_for_guestregister
+{
+    my ($self, %args) = @_;
+    $args{timeout} //= 300;
+    my $start_time = time();
```

Comment:
we can into a trouble easily if we will rely on absolute time. keep in mind overloaded workers on osd which can lead to situation when loop is in freeze but time is ticking so we run out of time without actually checking something